workos/event_objects/directory.py

```diff
@@ -0,0 +1,216 @@
+
+
+class DirectoryType(Enum):
+    AZURE_SCIM_v2 = "azure scim v2.0"
+    BAMBOO_HR = "bamboohr"
+    BREATHE_HR = "breathe hr"
+    CEZANNE_HR = "cezanne hr"
+    CYBERARK_SCIM_v2 = "cyperark scim v2.0"
+    FOURTH_HR = "fourth hr"
+    GENERIC_SCIM_v2 = "generic scim v2.0"
+    GOOGLE = "gsuite directory"
+    HIBOB = "hibob"
+    JUMPCLOUD_SCIM_v2 = "jump cloud scim v2.0"
+    OKTA_SCIM_v2 = "okta scim v2.0"
+    ONELOGIN_SCIM_v2 = "onelogin scim v2.0"
+    PEOPLE_HR = "people hr"
+    PERSONIO = "personio"
+    PINGFEDERATE_SCIM_v2 = "pingfederate scim v2.0"
+    RIPPLING_SCIM_v2 = "rippling v2.0"
+    SFTP = "sftp"
+    SFTP_WORKDAY = "sftp workday"
+    WORKDAY = "workday"
+
+
+class DirectoryState(Enum):
+    ACTIVE = "active"
+    DELETING = "deleting"
+
+
+class OrganizationDomain(TypedDict):
+    id: str
+    domain: str
+
+
+class WorkOSOrganizationDomain:
+    id: str
+    domain: str
+
+    @classmethod
+    def construct_from_response(cls, response: dict):
+        instance = cls()
+        for k, v in response.items():
+            setattr(instance, k, v)
+
+        return instance
+
+    def to_dict(self) -> OrganizationDomain:
+        return self.__dict__
+
+    def __str__(self):
+        return f"{self.__class__.__name__} {self.to_dict()}"
+
+
+class DirectoryEventWithLegacyFields(TypedDict):
+    id: str
+    name: str
+    type: DirectoryType
+    state: DirectoryState
+    domains: List[OrganizationDomain]
+    organization_id: str
+    created_at: str
+    updated_at: str
+    external_key: str
+    object: Literal["directory"]
+
+
+class WorkOSDirectoryEventWithLegacyFields:
+    id: str
+    name: str
+    type: DirectoryType
+    state: DirectoryState
+    domains: List[WorkOSOrganizationDomain]
+    organization_id: str
+    created_at: str
+    updated_at: str
+    external_key: str
+    object: Literal["directory"]
+
+    @classmethod
+    def construct_from_response(cls, response: dict):
+        instance = cls()
+        for k, v in response.items():
+            if k == "domains":
+                domains = []
+                for domain in v:
+                    domains.append(
+                        WorkOSOrganizationDomain.construct_from_response(domain)
+                    )
+                setattr(instance, k, domains)
+            elif k == "type":
+                setattr(instance, k, DirectoryType(v))
+            elif k == "state":
+                setattr(instance, k, DirectoryState(v))
+            else:
+                setattr(instance, k, v)
+
+        return instance
+
+    def to_dict(self) -> DirectoryEventWithLegacyFields:
+        return self.__dict__
+
+    def __str__(self):
+        return f"{self.__class__.__name__} {self.to_dict()}"
+
+
+class DirectoryEvent:
+    id: str
+    name: str
+    type: DirectoryType
+    state: DirectoryState
+    organization_id: str
+    created_at: str
+    updated_at: str
+    object: Literal["directory"]
+
+
+class WorkOSDirectoryEvent:
```

Comment:
The naming here is a little confusing. This isn't an event, it's the data/payload. Later on you also define an event class that actually has the `id`, `event`, and `created_at` keys from the event.
```suggestion
class WorkOSDirectoryEventData:
```